crates/distribution-filename/src/extension.rs

```diff
@@ -68,17 +68,18 @@ impl SourceDistExtension {
         };
 
         match extension {
-            "zip" => Ok(Self::Zip),
+            "zip"=> Ok(Self::Zip),
             "gz" if is_tar(path.as_ref()) => Ok(Self::TarGz),
-            "tgz" => Ok(Self::TarGz),
-            "bz2" if is_tar(path.as_ref()) => Ok(Self::TarBz2),
-            "xz" if is_tar(path.as_ref()) => Ok(Self::TarXz),
+            "tgz" | "tar" => Ok(Self::TarGz), // `.tar` included here
```

Comment:
It might be that `.tar` is just uncompressed tar without gzip?

Reply:
Thanks for suggestion.

Reply:
Should I need to update the SourceDistExtension enum?

Reply:
I believe so, yeah.

Reply:
Do you want some help finishing this off?

Reply:
Yeah I need some help, 

in extension.rs file: 
```
pub enum SourceDistExtension {
    Zip,
    TarGz,
    TarBz2,
    TarXz,
    Tar,
    TarZst,
    TarLz,
    TarLzma,
}
```

```
match extension {
            "zip" => Ok(Self::Zip),
            "gz" if is_tar(path.as_ref()) => Ok(Self::TarGz),
            "tgz" => Ok(Self::TarGz),
            "tar" => Ok(Self::Tar),
            "bz2" | "tbz" if is_tar(path.as_ref()) => Ok(Self::TarBz2),
            "xz" | "txz" | "tlz" | "tar.lz" | "tar.lzma" if is_tar(path.as_ref()) => {
                Ok(Self::TarXz)
            }
            "zst" if is_tar(path.as_ref()) => Ok(Self::TarZst),
            _ => Err(ExtensionError::SourceDist),
        }
```
```
impl Display for SourceDistExtension {
    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
        match self {
            Self::Zip => f.write_str("zip"),
            Self::TarGz => f.write_str("tar.gz"),
            Self::TarBz2 => f.write_str("tar.bz2"),
            Self::TarXz => f.write_str("tar.xz"),
            Self::Tar => f.write_str("tar"),
            Self::TarZst => f.write_str("tar.zst"),
            Self::TarLz => f.write_str("tar.lz"),
            Self::TarLzma => f.write_str("tar.lzma"),
        }
    }
}

// #[derive(Error, Debug)]
pub enum ExtensionError {
    #[error("`.whl`, `.zip`, `.tar.gz`, `.tgz`, `.tar.bz2`, `.tbz`, `.tar.xz`, `.txz`, `.tar.lz`, `.tar.lzma`, `.tar.zst`")]
    Dist,
    #[error("`.zip`, `.tar.gz`, `.tgz`, `.tar.bz2`, `.tbz`, `.tar.xz`, `.txz`, `.tar.lz`, `.tar.lzma`, `.tar.zst`")]
    SourceDist,
}

```

Where am I going wrong, as some tests are being failed. I tried to took inspiration from the issue #7201 as he done the same thing.